code/SDL2/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+if (MSVC)
+	set(SDL2_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/include CACHE STRING "")
```

Comment:
In WorldOfPadman we have solved this like this: https://github.com/PadWorld-Entertainment/worldofpadman/blob/main/libs/CMakeLists.txt#L10

```cmake
if (UNIX)
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(SDL2 REQUIRED sdl2)
    add_library(SDL2_system INTERFACE)
    add_library(SDL2_main INTERFACE)
    target_include_directories(SDL2_system INTERFACE ${SDL2_INCLUDE_DIRS})
    target_link_libraries(SDL2_system INTERFACE ${SDL2_LINK_LIBRARIES})
    add_library(SDL2::SDL2 ALIAS SDL2_system)
    add_library(SDL2::SDL2main ALIAS SDL2_main)
else()
    set(OUTPUT_DIRECTORY "${ROOT_DIR}/${ENGINE_BINARY_DIR}")
    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG "${OUTPUT_DIRECTORY}")
    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE "${OUTPUT_DIRECTORY}")
    set(CMAKE_LIBRARY_OUTPUT_DIRECTORY_DEBUG "${OUTPUT_DIRECTORY}")
    set(CMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE "${OUTPUT_DIRECTORY}")
    set(SDL_CMAKE_DEBUG_POSTFIX "" CACHE STRING "Name suffix for debug builds")
    add_subdirectory(SDL2)
endif()
```

We have an SDL2 clone in our repository - which is a lot easier to update and maintain than building the needed libs each time.